beacon-chain/core/validators/validator_test.go

```diff
@@ -398,6 +398,60 @@ func TestActivateValidator_OK(t *testing.T) {
 	}
 }
 
+func TestDoubleActivatedValidator(t *testing.T) {
+	state := &pb.BeaconState{
+		Slot: 100, // epoch 2
+		ValidatorRegistry: []*pb.Validator{
+			{Pubkey: []byte{'A'}},
+		},
+	}
+	newState, err := ActivateValidator(state, 0, false)
+	if err != nil {
+		t.Fatalf("could not execute activateValidator:%v", err)
+	}
+	currentEpoch := helpers.CurrentEpoch(state)
```

Comment:
this is a direct copy of `updateRegistry`, this test will not capture any regression. You are supposed to test the core functions not copy and paste the code into the test

Reply:
+1

Reply:
I thought it would work, but clearly no lines changed. I'll keep making fixes until it's right. I'll add a comment once I get better test coverage.